docs/examples/ambassador-ingress.md

```diff
@@ -1,186 +1,181 @@
+
+## Use Docker for non-native k0s platforms
+
+With Docker you can run k0s on platforms that the distribution does not
+natively support (refer to [Run k0s in Docker](../k0s-in-docker.md)). Skip this
+section if you are on a platform that k0s natively supports.
+
+As you need to create a custom configuration file to install Ambassador
+Gateway, you will first need prepare to map that file into the k0s container
+and to expose the ports Ambassador needs for outside access.
+
+1. Run k0s under Docker:
+
+    ```sh
+    docker run -d --name k0s --hostname k0s --privileged -v /var/lib/k0s -p 6443:6443 docker.io/k0sproject/k0s:latest
+    ```
+
+2. Export the default k0s configuration file:
+
+    ```sh
+    docker exec k0s k0s default-config > k0s.yaml
+    ```
+
+3. Export the cluster config, so you can access it using kubectl:
+
+    ```sh
+    docker exec k0s cat /var/lib/k0s/pki/admin.conf > k0s-cluster.conf
+    export KUBECONFIG=$KUBECONFIG:<absolute path to k0s-cluster.conf>
+
+## Configure k0s.yaml for Ambassador Gateway
+
+1. Open the ``k0s.yml`` file and append the following extensions at the end:
+
+    ```yaml
+    extensions:
+      helm:
+        repositories:
+          - name: datawire
+            url: https://www.getambassador.io
+          - name: bitnami
+            url: https://charts.bitnami.com/bitnami
+        charts:
+          - name: ambassador
+            chartname: datawire/ambassador
+            version: "6.5.13"
+            namespace: ambassador
+            values: |2
+              service:
+                externalIPs:
+                - 172.17.0.2
+          - name: metallb
+            chartname: bitnami/metallb
+            version: "1.0.1"
+            namespace: default
+            values: |2
+              configInline:
+                address-pools:
+                - name: generic-cluster-pool
+                  protocol: layer2
+                  addresses:
+                  - 172.17.0.2
+    ```
+
+    **Note**: It may be necessary to replace the 172.17.0.2 IP with your local IP address.
+
+    This action adds both Ambassador and Metallb (required for LoadBalancers) with the corresponding repositories and (minimal) configurations. Be aware that the provided example illustrates the use of your local network and that you will want to provide a range of IPs for MetalLB that are addressable on your LAN to access these services from anywhere on your network.
+
+2. Stop/remove your k0s container:
+
+    ```sh
+    docker stop k0s
+    docker rm k0s
+    ```
+
+3. Retart your k0s container, this time with additional ports and the above config file mapped into it:
+
+      ```sh
+      docker run --name k0s --hostname k0s --privileged -v /var/lib/k0s -v <path to k0s.yaml file>:/k0s.yaml -p 6443:6443 -p 80:80 -p 443:443 -p 8080:8080 docker.io/k0sproject/k0s:latest
+      ```
+
+    After some time, you will be able to list the Ambassador Services:
+
+    ```shell
+    kubectl get services -n ambassador
```

Comment:
I would prefix the shell examples with a dollar sign, like:

```shell
$ kubectl get services
```

it's easier to see which rows are the entered commands and which are the command output. I think it's quite common in such examples.

Reply:
Thanks, Kimmo. Following our Slack discussion, I will resolve to NOT include the $ (for copy-paste purpose) and to instead separate the commands from the output (with a short text lead-in, as per Google standards). Fixing this one, but will do a sweep later in a separate PR for all.

Reply:
Sounds good to NOT include the $ (for copy-paste purpose)